### PR TITLE
RESTClient: add an example for creating a custom POST paginator

### PR DESCRIPTION
**Pull request by burnash**

### CODE REVIEW

File: docs/website/docs/general-usage/http/rest-client.md [hunk 1]
```diff
@@ -354,6 +354,38 @@ def get_data():
...
+            self._has_next_page = False
+        else:
+            self.cursor = response.json().get("next_page_cursor")
+
+    def update_request(self, request: Request) -> None:
+        if request.json is None:
+            request.json = {}
+
+        # Add the cursor to the request body
+        request.json["cursor"] = self.cursor
```
Comment (VioletM) [inline docs/website/docs/general-usage/http/rest-client.md:386]: Let's add a usage as well:

```suggestion
        request.json["cursor"] = self.cursor
       
client = RESTClient(
    base_url="https://api.example.com",
    paginator= PostBodyPaginator()
)

@dlt.resource
def get_data():
    for page in client.paginate("/data"):
        yield page
```

File: docs/website/docs/general-usage/http/rest-client.md [hunk 2]
```diff
@@ -354,6 +354,38 @@ def get_data():
...
+[`PageNumberPaginator`](#pagenumberpaginator) that ships with dlt does the same thing, but with more flexibility and error handling. This example is meant to demonstrate how to implement a custom paginator. For most use cases, you should use the [built-in paginators](#paginators).
+:::
+
+#### Example 2: creating a paginator for POST requests
+
+Some APIs use POST requests for pagination, where the next page is fetched by sending a POST request with a cursor or other parameters in the request body. This is frequently used in "search" API endpoints or other endpoints with big payloads. Here's how you could implement a paginator for a case like this:
+
+```py
+from dlt.sources.helpers.rest_client.paginators import BasePaginator
+from dlt.sources.helpers.requests import Response, Request
```
Comment (VioletM) [inline docs/website/docs/general-usage/http/rest-client.md:367]:
```suggestion
from dlt.sources.helpers.rest_client import RESTClient
from dlt.sources.helpers.requests import Response, Request
```

File: docs/website/docs/general-usage/http/rest-client.md [hunk 3]
```diff
@@ -311,7 +311,7 @@ When working with APIs that use non-standard pagination schemes, or when you nee
 
 - `update_request(request: Request) -> None`: Before making the next API call in `RESTClient.paginate` method, `update_request` is used to modify the request with the necessary parameters to fetch the next page (based on the current state of the paginator). For example, you can add query parameters to the request, or modify the URL.
 
-#### Example: creating a query parameter paginator
+#### Example 1: creating a query parameter paginator
```
Comment (VioletM) [inline docs/website/docs/general-usage/http/rest-client.md:314]: NIT cosmetics
```suggestion
#### Example 1: Creating a query parameter paginator
```

File: docs/website/docs/general-usage/http/rest-client.md [hunk 4]
```diff
@@ -354,6 +354,38 @@ def get_data():
         yield page
 ```
 
+:::tip
+[`PageNumberPaginator`](#pagenumberpaginator) that ships with dlt does the same thing, but with more flexibility and error handling. This example is meant to demonstrate how to implement a custom paginator. For most use cases, you should use the [built-in paginators](#paginators).
+:::
+
+#### Example 2: creating a paginator for POST requests
```
Comment (VioletM) [inline docs/website/docs/general-usage/http/rest-client.md:361]: NIT cosmetics
```suggestion
#### Example 2: Creating a paginator for POST requests
```

**Review by VioletM — DISMISSED** [review]
It will be very helpful for people!

File: docs/website/docs/general-usage/http/rest-client.md (hunk 3 above)
Comment (burnash) [reply docs/website/docs/general-usage/http/rest-client.md:314]: We're using the sentence case for all headings, so I think it should start with a lowercase after ":". Or not?

File: docs/website/docs/general-usage/http/rest-client.md (hunk 2 above)
Comment (burnash) [reply docs/website/docs/general-usage/http/rest-client.md:367]: Nice catch!

File: docs/website/docs/general-usage/http/rest-client.md [hunk 5]
```diff
@@ -354,6 +354,49 @@ def get_data():
...
+    def update_request(self, request: Request) -> None:
+        if request.json is None:
+            request.json = {}
+
+        # Add the cursor to the request body
+        request.json["cursor"] = self.cursor
+       
+client = RESTClient(
+    base_url="https://api.example.com",
+    paginator= PostBodyPaginator()
```
Comment (burnash) [inline docs/website/docs/general-usage/http/rest-client.md:391]:
```suggestion
    paginator=PostBodyPaginator()
```

**Review by rudolfix — APPROVED** [review]

**Pull request merged**